File: Data/Werk.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MaestroNotes.Data
{
    public class Werk
    {
        [Key]
        public int Id { get; set; }
        [MaxLength(200)]
        public string Name { get; set; } = "";
        [MaxLength(1000)]
        public string? Note { get; set; }

        public int? KomponistId { get; set; }
        [ForeignKey("KomponistId")]
        public Komponist? Komponist { get; set; }

        public List<MusicRecord> MusicRecords { get; set; } = new();
    }
}

File: php-app/app/Models/Werk.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Werk extends Model
{
    use HasFactory;

    protected $table = 'werke';

    protected $fillable = [
        'name',
        'note',
        'komponist_id',
    ];

    public function komponist()
    {
        return $this->belongsTo(Komponist::class);
    }

    public function musicRecords()
    {
        return $this->belongsToMany(MusicRecord::class, 'music_record_werk');
    }
}
